api/rest/client/client.go

```diff
@@ -164,6 +178,52 @@ func (c *Client) setupHostname() error {
 		}
 	default:
 		c.hostname = fmt.Sprintf("%s:%s", c.config.Host, c.config.Port)
+		apiAddr, err := ma.NewMultiaddr(
+			fmt.Sprintf("/ip4/%s/tcp/%s", c.config.Host, c.config.Port),
+		)
+		if err != nil {
+			return err
+		}
+		c.config.APIAddr = apiAddr
 	}
 	return nil
 }
+
+func (c *Client) setupProxy() error {
+	if c.config.ProxyAddr != nil {
+		return nil
+	}
+
+	// Guess location from PeerAddr or APIAddr
+	port, err := ma.NewMultiaddr(fmt.Sprintf("/tcp/%d", DefaultProxyPort))
+	if err != nil {
+		return err
+	}
+	var paddr ma.Multiaddr
+	switch {
+	case c.config.PeerAddr != nil:
+		paddr = ma.Split(c.config.PeerAddr)[0].Encapsulate(port)
+	case c.config.APIAddr != nil: // Host/Port setupHostname sets APIAddr
+		paddr = ma.Split(c.config.APIAddr)[0].Encapsulate(port)
+	default:
+		return errors.New("cannot guess default proxy address")
```

Comment:
This message throws me off a bit.  Are we saying that there are many potential defaults that we tried to guess and we can't finish guessing, or are we saying that there is no default and so there is no guessing to be done?  When I read this I wonder "why can't it guess the default, surely the code knows what the default value is?"

Perhaps "cannot find proxy address"?

Reply:
ok. This should be dead code, as when setupProxy() either PeerAddr or APIAddr will be set (or something will have errored before.